provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/CachingKotlinCompiler.kt

```diff
@@ -118,8 +118,8 @@ class CachingKotlinCompiler(
     }
 
     private
-    fun scriptFileNameFor(scriptPath: String) =
-        scriptPath.substringAfterLast(File.separatorChar)
+    fun scriptFileNameFor(scriptTemplate: KClass<*>, scriptPath: String) =
+        "${scriptTemplate.qualifiedName}_${scriptPath.substringAfterLast(File.separatorChar)}"
```

Comment:
:x: `scriptTemplate.qualifiedName` should be part of the cache key but not necessarily the file name, right?

Reply:
Right, done